### PR TITLE
Remove deprecated testcase implementation

### PR DESCRIPTION
**Pull request by samnela**

### CODE REVIEW

**Comment by samnela** [comment]
Build failed on unrelated reason

**Comment by mbonneau** [comment]
@samnela Thanks for the PR! I am going to see if I can fix that unrelated test and get this merged.

File: tests/Integration/ConnectionTest.php [hunk 1]
```diff
@@ -129,7 +129,9 @@ function () {
 
         $this->runLoopWithTimeout(2);
 
-        $this->assertInstanceOf(RecordNotFoundException::class, $error);
+        // At some point, DNS was returning RecordNotFoundException
+        // as long as we are getting an Exception here, we should be good
+        $this->assertInstanceOf(\Exception::class, $error);
```
Comment (CharlotteDunois) [inline tests/Integration/ConnectionTest.php]: The `RecordNotFoundException` is now the previous exception of the "returned" `RuntimeException`. Use `$error->getPrevious()` to get the `RecordNotFoundException`.

**Comment by samnela** [comment]
Thank you @mbonneau

**Pull request merged**